### PR TITLE
Markdown doc for `ErrorOutOfGasSloadSstore` state

### PR DESCRIPTION
**Pull request by silathdiir**
Upstream circuit issue https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1119

Circuit PR https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1147

### CODE REVIEW

File: specs/error_state/ErrorOutOfGasSLOAD.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+For this gadget, the gas cost is calculated with [EIP-2929] as:
+```
+gas_cost = GAS_COST_WARM_ACCESS if is_warm_access else COLD_SLOAD_COST
+```
+
+### Constraints
+
+1. Constrain `gas_left < gas_cost`.
+2. Current call must fail.
+3. If it's a root call, it transits to `EndTx`.
```
Comment (DreamWuGit) [inline specs/error_state/ErrorOutOfGasSLOAD.md:24]: add rw_counter_of_reversion_contraint here .

File: specs/error_state/ErrorOutOfGasSLOAD.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+4. if it isn't a root call, it restores caller's context by reading to `rw_table`, then does step state transition to it.
+
+### Lookups
+
+1. 4 call context lookups for `tx_id`, `callee_address`, `is_success` and `rw_counter_end_of_reversion`.
+2. 1 stack read for `storage_key` (no stack write for `storage_value` when out of gas).
+3. 1 account storage access list read for `storage_key`.
+
+[//]: # "badges"
+[go-ethereum]: https://github.com/ethereum/go-ethereum
```
Comment (DreamWuGit) [inline specs/error_state/ErrorOutOfGasSLOAD.md:34]: remove these links ?

Comment (silathdiir) [reply specs/error_state/ErrorOutOfGasSLOAD.md]: Deleted links in [aa8d6f1](https://github.com/scroll-tech/zkevm-specs/pull/57/commits/aa8d6f1ec6490a6129603bbe7a06a8c9b7d5ace5).

File: specs/error_state/ErrorOutOfGasSLOAD.md (hunk 1 above)
Comment (silathdiir) [reply specs/error_state/ErrorOutOfGasSLOAD.md:24]: Added `rw_counter_end_of_reversion = rw_counter_end_of_step + reversible_counter` constraint in [99df3b7](https://github.com/scroll-tech/zkevm-specs/pull/57/commits/99df3b7196c297ae547833c55d8f5664fdeaf5bd).

**Comment by silathdiir** [comment]
Thanks for @DreamWuGit 's code review. Moved to upstream https://github.com/privacy-scaling-explorations/zkevm-specs/pull/381

**Pull request closed without merging**